src/utils/common.js

```diff
@@ -17,4 +17,28 @@ const classNames = (names) => {
 
 }
 
-export { classNames };
\ No newline at end of file
+const webShare = (data) => {
+    if (navigator.share) { 
+        navigator.share({
+           title: data.title,
+           text: data.text,
+           url: data.url
+        }).then(() => {
+            if (typeof data.successCallback === 'function') {
```

Comment:
why not return promise (i.e. make `webShare` async function) instead of using callbacks, especially given that `navigator.share` is already returning a promise?

Reply:
What do you think about this?

In `webShare.js`: 
```
class NoSupportError extends Error {}

const webShare = (data) => {
    if (navigator.share) { 
        return navigator.share({
           title: data.title,
           text: data.text,
           url: data.url
        })
    } else {
        return Promise.reject(new NoSupportError());
    }
}
```

Usage:

```
webShare({
            url: window.location.href,
}).catch(err => {
            if (err instanceof NoSupportError) {
                this.handleCopyDesktop();
            } else {
                console.error(err);
                this.handleCopyDesktop();
            }
});
```